src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -49,9 +48,15 @@ trait RegularSync {
       waitingForActor = None
       handleBlockBodies(peer, blockBodies)
 
-    case block: BroadcastBlocks if broadcasting =>
+    case BroadcastBlocks(newBlocks) if broadcasting =>
       //FIXME: Decide block propagation algorithm (for now we send block to every peer) [EC-87]
-      peersToDownloadFrom.keys.foreach(_.ref ! block)
+      val blocksToSendToEachPeer = for {
+        peer <- peersToDownloadFrom.keys
```

Comment:
Although I think having a `MessageToPeers` it's a nice solution, as adding the `PeerConnection` event (and the `PeerStatusUpdated` event for when the peer status is updated) are needed for not asking periodically for the peers in the `SyncController`, maybe we could take advantage of that to solve the problem you found as well?